zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/approval/ApproveTask.java

```diff
@@ -492,13 +492,20 @@ protected String filePathGenerator(String datasetId, Date publishDate, String Da
         String finalPath;
 
         if (nonTsDatasetWhitelistSet.contains(baseFilename)) {
-            finalPath = "ts-datasets/" + "other" + "/" + formattedDate;
+            finalPath = "ts-datasets/" + "other" + "/" + formattedDate + "/" + pickCorrectDatasetVersion(correctDatasetVersion, DatasetVersion);
         } else {
-            finalPath = "ts-datasets/" + baseFilename + "/" + (!correctDatasetVersion.equals("")? incrementDatasetVersionByOne(correctDatasetVersion) : DatasetVersion);
+            finalPath = "ts-datasets/" + baseFilename + "/" + pickCorrectDatasetVersion(correctDatasetVersion, DatasetVersion);
         }
         return finalPath;
     }
 
+    protected String pickCorrectDatasetVersion(String correctDatasetVersion, String datasetVersion) {
+        if (datasetVersion == null || correctDatasetVersion == null) {
+            throw new IllegalArgumentException("input can't be null");
```

Comment:
You're not declaring this 'throw' which will mean that your parent method will also fall over etc.